core/src/main/java/io/questdb/griffin/engine/table/ShowCreateTableRecordCursorFactory.java

```diff
@@ -0,0 +1,273 @@
+    }
+
+    public static class ShowCreateTableCursor implements NoRandomAccessRecordCursor {
+        private final ShowCreateTableRecord record = new ShowCreateTableRecord();
+        private final Utf8StringSink sink = new Utf8StringSink();
+
+        private SqlExecutionContext executionContext;
+        private boolean hasRun;
+        private CairoTable table;
+        private TableToken tableToken;
+
+        @Override
+        public void close() {
+            sink.clear();
+        }
+
+        @Override
+        public Record getRecord() {
+            return record;
+        }
+
+        @Override
+        public boolean hasNext() {
+            if (!hasRun) {
+                sink.clear();
+                // CREATE TABLE table_name
+                sink.put("CREATE TABLE ")
+                        .put(tableToken.getTableName())
+                        .put(" ( ")
+                        .put('\n');
+
+                // column_name TYPE
+                for (int i = 0, n = (int) table.getColumnCount(); i < n; i++) {
+                    final CairoColumn column = table.getColumnQuiet(i);
+                    sink.put('\t')
+                            .put(column.getName())
+                            .put(' ')
+                            .put(ColumnType.nameOf(column.getType()));
+
+                    if (column.getType() == ColumnType.SYMBOL) {
+                        // CAPACITY value (NO)CACHE
+                        sink.put(" CAPACITY ").put(column.getSymbolCapacity());
+                        sink.put(column.getSymbolCached() ? " CACHE" : " NOCACHE");
+
+                        if (column.getIsIndexed()) {
+                            // INDEX CAPACITY value
+                            sink.put(" INDEX CAPACITY ")
+                                    .put(column.getIndexBlockCapacity());
+                        }
+                    }
+
+                    if (i < n - 1) {
+                        sink.put(',');
+                    }
+                    sink.put('\n');
+                }
+                sink.put(')');
+
+                // timestamp(ts)
+                if (table.getTimestampIndex() != -1) {
+                    sink.put(" timestamp(")
+                            .put(table.getTimestampName())
+                            .put(')');
+
+                    // PARTITION BY unit
+                    if (table.getPartitionBy() != PartitionBy.NONE) {
+                        sink.put(" PARTITION BY ").put(table.getPartitionByName());
+                    }
+
+                    // (BYPASS) WAL
+                    if (!table.getWalEnabled()) {
+                        sink.put(" BYPASS");
+                    }
+                    sink.put(" WAL");
+                }
+
+                final CairoConfiguration config = executionContext.getCairoEngine().getConfiguration();
+                final boolean withMaxUncommittedRows = table.getMaxUncommittedRows() != config.getMaxUncommittedRows();
```

Comment:
this could lead to undesired side-effects, consider we asking DDL from a user, who's defaults coincide with the config. DDL is produced without explicit values to us. We run DDL with our local defaults - cannot repro problem